src/components/location/location.service.ts

```diff
@@ -205,6 +205,7 @@ export class LocationService {
         `
           MATCH (location {active: true})-[:name {active:true }]->(name:Property {active: true})
           WHERE name.value CONTAINS $filter
+          AND ('Zone' IN labels(location) OR 'Region' IN labels(location) OR 'Country' IN labels(location))
```

Comment:
I'm not going to harp on this right now, but the base node should have a `Location` label. That way we can select it in the match clause instead of having to filter here.

Reply:
If we don't use this clause, it will fetch all nodes which `Property.name` contains `filter` and call `readOne` function. And in this `readOne` function if the label doesn't match with [`Country`, `Region`, `Zone`], it will throw `BadRequest`.. so I added this clause in this query.

Reply:
We will always need labels on match clauses to prevent a full DB scan. Labels are indexed.

Reply:
`UNION` may be worth looking into. You can make 3 separate queries using labels that should provide better performance and do what you need it to.

Reply:
`UNION` shouldn't be necessary. The base node can be created with all 3 labels, `BaseNode:Location:Country` and then we match for whatever the use case is.

Reply:
that's right, forgot about :Location being a generic